src/control.ts

```diff
@@ -0,0 +1,73 @@
+// I don't love this, but it seems to do the job.
```

Comment:
A simpler approach that I think achieves the same thing makes use of Typescript's 'Partial' keyword:
1. Make all properties in `DopriControlParam` non-nullable
2. Ditch `DopriControl` and `dopriControl`
3. Change the `Dopri` constructor as follows:

```

    constructor(rhs: RhsFn, n: number, controlOptions: Partial<DopriControlParam> = {}) {
        this._stepper = new dopri5.Dopri5(rhs, n);

        const defaults = {maxSteps: 10000, atol: 1e-6, rtol: 1e-6, stiffCheck: 0};

        this._control = {
            maxSteps: controlOptions.maxSteps || defaults.maxSteps,
            stiffCheck: controlOptions.stiffCheck || defaults.stiffCheck,
            rtol: controlOptions.rtol || defaults.rtol,
            atol: controlOptions.atol || defaults.atol
        };

        if (this._control.maxSteps < 1) {
            throw controlError("maxSteps", "must be at least 1");
        }
        if (this._control.atol <= 0) {
            throw controlError("atol", "must be strictly positive");
        }
        if (this._control.rtol <= 0) {
            throw controlError("rtol", "must be strictly positive");
        }
    }
```